Sources/SCTE35/HelperConverter/BitConverter.swift

```diff
@@ -87,56 +166,168 @@ class BitConverter {
         return stringToReturn
     }
 
+    static func bits(fromUMID umid: String) -> [Bit]? {
+        guard umid.replacingOccurrences(of: ".", with: "").count == 64 else { return nil}
+
+        let hexParts = umid.split(separator: ".")
+        guard hexParts.count == 8 else { return nil }
+
+        var bits = [Bit]()
+        for hexChar in hexParts {
+            guard let integer = UInt32(hexChar, radix: 16) else {
+                return nil
+            }
+            bits.append(contentsOf: BitConverter.bits(from: integer))
+        }
+        return bits
+    }
+
     static func isanString(fromBits bits: [Bit]) -> String? {
-        guard bits.count == 96 else { return nil }
-        let nibbles = stride(from: 0, to: bits.count, by: 4).map { (last) -> [Bit] in
-            return Array(bits[last..<last+4])
+        let containsVersion: Bool
```

Comment:
The ISAN string from this method was incomplete.  Added support for ISAN and VISAN as well as calculating check characters so that this function can return a fully correct ISAN string.